In [14]:
import openai
import datetime
import os

In [19]:


# Initialize the OpenAI client with your API key
openai.api_key = os.getenv('OPENAI_API_KEY')

# Open the audio file (replace with your file path)
audio_file_path = "./audio_english.wav"


pip install git+https://github.com/openai/whisper.git

sudo apt update && sudo apt install ffmpeg


In [18]:
print(datetime.datetime.now())
!whisper "audio_english.wav"
print(datetime.datetime.now())  



2024-03-11 22:29:35.371744
Detecting language using up to the first 30 seconds. Use `--language` to specify the language
Detected language: English
[00:00.000 --> 00:05.120]  Let us explore some basic examples of speech-to-text conversion in Python.
[00:05.760 --> 00:10.240]  There are a few libraries and tools you can use to achieve this.
[00:10.240 --> 00:14.080]  I'll provide examples using two popular options,
[00:14.080 --> 00:17.360]  speech recognition and OpenAI's whisper.
2024-03-11 22:29:43.674290


In [20]:
print(datetime.datetime.now())
!whisper "audio_german.wav" --language=de
print(datetime.datetime.now())  

2024-03-11 22:30:12.135949
[00:00.000 --> 00:06.440]  Während der US-Hersteller Boeing wegen massiver Sicherheitsmängel in die Krise stürzt, setzen
[00:06.440 --> 00:11.320]  Airbus und der brasilianische Konkurrent Embraer zum Höhenflug an.
[00:11.320 --> 00:13.840]  Was machen sie besser als der Erzrivale?
2024-03-11 22:30:20.313396


REsult is kind of slow with 8 and 9 seconds, but text transcription is excellent.

English:  
2024-03-11 22:29:35.371744  
Detecting language using up to the first 30 seconds. Use `--language` to specify the language  
Detected language: English  
[00:00.000 --> 00:05.120]  Let us explore some basic examples of speech-to-text conversion in Python.  
[00:05.760 --> 00:10.240]  There are a few libraries and tools you can use to achieve this.  
[00:10.240 --> 00:14.080]  I'll provide examples using two popular options,  
[00:14.080 --> 00:17.360]  speech recognition and OpenAI's whisper.  
2024-03-11 22:29:43.674290  

German:
2024-03-11 22:30:12.135949  
[00:00.000 --> 00:06.440]  Während der US-Hersteller Boeing wegen massiver Sicherheitsmängel in die Krise stürzt, setzen  
[00:06.440 --> 00:11.320]  Airbus und der brasilianische Konkurrent Embraer zum Höhenflug an.  
[00:11.320 --> 00:13.840]  Was machen sie besser als der Erzrivale?  
2024-03-11 22:30:20.313396  
